### ISSUE TITLE
Docker: is dotenv necessary in production?

### ISSUE THREAD

**Issue by albertsun** [cantino/huginn#1713]
Not really a bug, but a bit of confusion around how quoted environment variables are handled.

I'm trying to set the timezone to `TIMEZONE="Eastern Time (US & Canada)"`.

This doesn't work.
```
docker run --rm -it -p 3000:3000 -e TIMEZONE="Eastern Time (US & Canada)" cantino/huginn
/app/.env: line 66: syntax error near unexpected token `('
```

This does, sort of
```
docker run --name huginn -it -p 3000:3000 -e TIMEZONE="\"Eastern Time (US & Canada)\"" -d cantino/huginn
```

It runs, but now the `.env` file and the actual environment variable have different values.
```
 $ docker exec -it huginn bash
root@44a576493140:/app# cat .env | grep TIMEZONE
TIMEZONE="Eastern Time (US & Canada)"
root@44a576493140:/app# irb
irb(main):001:0> ENV['TIMEZONE']
=> "\"Eastern Time (US & Canada)\""
```

So if while running inside the docker container you try to enter a rails console for debugging purposes you get this error:

```
root@44a576493140:/app# ./bin/rails c
/app/vendor/bundle/ruby/2.2.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/time/zones.rb:72:in `rescue in find_zone!': Invalid Timezone: "Eastern Time (US & Canada)" (ArgumentError)
```

I guess you have to `source /app.env` first there.

Seems like the confusion comes from here:
https://github.com/cantino/huginn/blob/master/docker/single-process/scripts/init#L23

Trying to parse that, maybe there's an easy way to adjust that bit to recognize when env variables need to be quoted before writing them into `/app/.env`. Or is that section really necessary? Defaults seem to be set throughout the code whereever ENV is used for the most part — could we make that the standard? Then the first version of setting the environment variable would work.

**Comment by dsander** [comment]
Yes that is annoying, it has bitten me too. I think we could remove the `.env` in the docker container, however the problem I see is to maintain backwards compatibility.
If we remove the `.env` I would assume the docker container (and the rails console) would work with `-e TIMEZONE="Eastern Time (US & Canada)"` but would fail with `-e TIMEZONE="\"Eastern Time (US & Canada)\""` what is currently needed. It might be possible to convert the old double escaped strings into "normal" ones.

**Comment by cantino** [comment]
@alias1 if you're working on Docker stuff, this might be worth looking at.

**Comment by daviddavo** [comment]
The only setting that worked for me (docker-compose.yml) was changing the timezone using `TZ: "Europe/Madrid"`. Setting `TIMEZONE: "Madrid"` didn't change the output of the `date` command.
